feat: consume md docs directly (#78)
aws-powertools/powertools-mcp: src/tools/fetchDocPage/README.md

> # FetchDocPage Tool

Reviewer: This is whole page is a great bit of documentation.

```txt
Input:  https://docs.powertools.aws.dev/lambda/python/2.1.0/core/logger/
Output: https://docs.powertools.aws.dev/lambda/python/2.1.0/core/logger/index.md
```

## Flow Diagram

```mermaid
graph TD
    A[Fetch Request] --> B{Validate URL}
    B -->|Invalid| C[Return Validation Error]
    B -->|Valid| D[Transform URL to Markdown]
    
    D --> E[Generate Cache Key]
    E --> F[Parallel ETag Check]
    
    F --> G[Get Cached ETag]
    F --> H[Get Remote ETag via HEAD]
    
    G --> I{Both ETags Available?}
    H --> I
    
    I -->|No Cached ETag| J[Fetch Remote Page]
    I -->|No Remote ETag| J
    I -->|Both Available| K{ETags Match?}
    
    K -->|Yes| L[Get Cached Content]
    K -->|No| J
    
    L --> R{Content Available?}
    R -->|Yes| S[Return Cached Content]
    R -->|No - Corrupted/Missing| J[Fetch Remote Page]
    
    J --> M{Fetch Successful?}
    M -->|No| N[Return Fetch Error]
    M -->|Yes| O[Store Content in Cache]
    
    O --> P[Store ETag in Cache]
    P --> Q[Return Markdown Content]
    S --> Q
    
    style A fill:#e1f5fe
    style Q fill:#e8f5e8
    style C fill:#ffebee
    style N fill:#ffebee
```

## Caching Strategy

> The `fetchDocPage` tool implements a sophisticated ETag-based caching system that provides optimal performance while ensuring content freshness.

Reviewer: ```suggestion
The `fetchDocPage` tool implements an ETag-based caching system that ensures content freshness but remains performant.
```
